gyp/pylib/gyp/easy_xml.py

```diff
@@ -121,7 +122,10 @@ def WriteXmlIfChanged(content, path, encoding="utf-8", pretty=False, win32=False
 
     default_encoding = locale.getdefaultlocale()[1]
     if default_encoding and default_encoding.upper() != encoding.upper():
-        xml_string = xml_string.encode(encoding)
+        if (sys.version_info[0] >= 3) and (sys.version_info[1] >= 7):
```

Comment:
This won’t work for future releases, e.g. 4.0.

Reply:
What do u mean under "this"?

Reply:
The conditional. If `sys.version_info[0] == 4` and `sys.version_info[1] == 0` the test will fail (since 4 is greater than 3 but 0 is less than 7).

The tuple based comparison @cclauss provided avoids this.

Reply:
may be this solution will be fine:
`(sys.version_info[0] + sys.version_info[1] * 0.1) > 3.7`